fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -0,0 +1,67 @@
+package org.mozilla.fenix.shopping.middleware
+
+import mozilla.components.browser.engine.gecko.shopping.GeckoProductAnalysis
+import mozilla.components.browser.engine.gecko.shopping.Highlight
+import mozilla.components.concept.engine.shopping.ProductAnalysis
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState.OptedIn.ProductReviewState
+
+/**
+ * Maps [ProductAnalysis] to [ProductReviewState].
+ */
+fun ProductAnalysis?.toProductReviewState(): ProductReviewState =
+    if (this == null) {
+        ProductReviewState.Error
+    } else {
+        when (this) {
+            is GeckoProductAnalysis -> toProductReview()
+            else -> ProductReviewState.Error
+        }
+    }
+
+private fun GeckoProductAnalysis.toProductReview(): ProductReviewState =
+    if (productId == null) {
+        ProductReviewState.Error
+    } else {
+        val mappedRating = adjustedRating.toFloatOrNull()
+        val mappedGrade = grade?.toGrade()
+        val mappedHighlights = highlights?.toHighlights()?.toSortedMap()
+
+        if (mappedGrade == null && mappedRating == null && mappedHighlights == null) {
+            ProductReviewState.NoAnalysisPresent
+        } else {
+            ProductReviewState.AnalysisPresent(
+                productId = productId!!,
+                reviewGrade = mappedGrade,
+                needsAnalysis = needsAnalysis,
+                adjustedRating = mappedRating,
+                productUrl = analysisURL!!,
```

Comment:
should `analysisURL` also be in the root `if` with `productID`, if they're both mandatory?

Reply:
Good question, my understanding of the API is that when there's product ID and one of the 3 (grade, rating and highlights) present, there would be a fakespot analysis url present in the API.